tests/integ/modin/frame/test_cache_result.py

```diff
@@ -125,35 +142,135 @@ def test_cache_result_dataframe_complex_correctness(
+            snow_df = perform_chained_operations(snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                snow_df, native_df
+            )
+
+        with SqlCounter(query_count=1, join_count=1):
+            snow_df = pd.DataFrame(simple_test_data).pivot_table(**pivot_kwargs)
+            cached_snow_df = cache_and_return_df(snow_df, inplace)
+
+        with SqlCounter(query_count=1, union_count=9):
+            cached_snow_df = perform_chained_operations(cached_snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                cached_snow_df, native_df
+            )
+
+    def test_cache_result_post_apply(self, inplace, simple_test_data):
+        native_df = perform_chained_operations(
+            native_pd.DataFrame(simple_test_data).apply(lambda x: x + x, axis=1),
+            native_pd,
+        )
+        with SqlCounter(query_count=6, union_count=9, udtf_count=1):
+            snow_df = pd.DataFrame(simple_test_data).apply(lambda x: x + x, axis=1)
+            repr(snow_df)
+            snow_df = perform_chained_operations(snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                snow_df, native_df
+            )
+
+        with SqlCounter(query_count=5, udtf_count=1):
+            snow_df = pd.DataFrame(simple_test_data).apply(lambda x: x + x, axis=1)
+            cached_snow_df = cache_and_return_df(snow_df, inplace)
+
+        with SqlCounter(query_count=2, union_count=9):
+            repr(cached_snow_df)
+            cached_snow_df = perform_chained_operations(cached_snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                cached_snow_df,
+                native_df,
+            )
+
+    def test_cache_result_post_applymap(self, inplace, simple_test_data):
+        native_df = perform_chained_operations(
+            native_pd.DataFrame(simple_test_data).applymap(lambda x: x + x), native_pd
+        )
+        with SqlCounter(
+            query_count=11,
+            union_count=9,
+            udf_count=2,
+            high_count_expected=True,
+            high_count_reason="applymap requires additional queries.",
+        ):
+            snow_df = pd.DataFrame(simple_test_data).applymap(lambda x: x + x)
+            repr(snow_df)
+            snow_df = perform_chained_operations(snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                snow_df, native_df
+            )
+
+        with SqlCounter(
+            query_count=10,
+            high_count_expected=True,
+            high_count_reason="caching result of udf seems to have high query count.",
+        ):
+            snow_df = pd.DataFrame(simple_test_data).applymap(lambda x: x + x)
+            cached_snow_df = cache_and_return_df(snow_df, inplace)
+
+        with SqlCounter(query_count=2, union_count=9, udf_count=0):
```

Comment:
maybe comment that one query comes from `repr` and the next from `assert_snowpark_pandas_equals_to_pandas_without_dtypecheck`.